loadtest/control/simulcontroller/utils.go

```diff
@@ -60,11 +60,22 @@ func splitName(name string) (string, string) {
 	return prefix, typed
 }
 
+func getCutoff(prefix, typed string, seed ...int64) int {
+	if len(seed) > 0 { // for deterministic test output
+		rand.Seed(seed[0])
+	}
```

Comment:
Sorry to jump in but this caught my attention. Do we actually need to pass the seed?
We usually rely on the default source which is set in `TestMain`.

Reply:
The seed set in `TestMain` is set using `time.Now().Unix()`. This means the seed changes depending on when the test runs, therefore giving us different random values at different times. 

To get deterministic values, I pass the seed as a constant in the tests (see [line 183](https://github.com/mattermost/mattermost-load-test-ng/blob/MM-25850-fix-random-number-usage/loadtest/control/simulcontroller/utils_test.go#L183) ) so that I will always get the same output regardless of when the test is run.

Reply:
The idea behind the seed was that it's supposed to be time based so that that tests can cover more cases and better simulate what running the main program would do. If a test ever failed we would be able to reproduce it through the `MM_LOADTEST_SEED` environment variable, effectively making it deterministic (at the test level anyway).

Reply:
Do you have a suggestion for how this change can be tested? Without a deterministic seed, we can't confirm the actual output of a random result. 
I suppose we could water down the test to just confirm that it does not panic instead of checking the actual value.

Reply:
Any reason we cannot set the seed in the tests themselves ?

Reply:
>Any reason we cannot set the seed in the tests themselves ?

Do you mean set `rand.Seed(1)` in the test itself? That's the first thing I tried, but I wasn't getting "deterministic" results. I'm not sure why.

Reply:
> Do you have a suggestion for how this change can be tested? Without a deterministic seed, we can't confirm the actual output of a random result.

@ashishbhate I apologize, I misread the test. I didn't notice you were relying on the actual output for seed 1. 

I think for this use case it would be more appropriate to pass a custom source created with `rand.NewSource()` instead of a seed value and if `nil` rely on the default one.

Setting the global seed in single tests is making things more difficult, we can't get back the original seed and printing a different seed for every test that makes use of one seems a bit impractical.

Reply:
That sounds good. I would suggest not to make this a variadic arg. That seems like we are trying to hide the fact from the user that a test passes a source, but the main code does not. Essentially this makes it act like a default variable which Go explicitly avoids. Maybe we can just pass nil explicitly. It's a compromise though. 1/5.

Reply:
>Setting the global seed in single tests is making things more difficult, we can't get back the original seed and printing a different seed for every test that makes use of one seems a bit impractical.

Yep, agreed. The current solution isn't great. I looked into patching `SetRandomSeed`, the config and the tests somehow to store the seed the original seed and then restore it, but that might not do exactly what we want and makes this patch very noisy.

>I think for this use case it would be more appropriate to pass a custom source created with rand.NewSource() instead of a seed value and if nil rely on the default one.

Sounds good, thanks for the input!

>...I would suggest not to make this a variadic arg. That seems like we are trying to hide the fact from the user that a test passes a source, but the main code does not. Essentially this makes it act like a default variable which Go explicitly avoids. It's a compromise though. 1/5.

I think its a reasonable way to implement optional arguments, especially for edge cases like this. It also saves us the trouble of having to change all usages. 

Since its a tie for and against, I will leave it in for now :) If someone else weighs in I will update it.